sentry_auth_google/views.py

```diff
@@ -16,27 +16,37 @@ def __init__(self, domain=None, *args, **kwargs):
     def dispatch(self, request, helper):
         access_token = helper.fetch_state('data')['access_token']
 
-        req = safe_urlopen('{0}?{1}'.format(
+        req = safe_urlopen('{0}?{1}&alt=json'.format(
             USER_DETAILS_ENDPOINT,
             urlencode({
                 'access_token': access_token,
-            }),
+            })
         ))
         body = safe_urlread(req)
         data = json.loads(body)
 
-        if not data.get('domain'):
-            return helper.error(ERR_INVALID_DOMAIN)
+        if not data.get('data'):
+            #TODO(cnuss): Propagate this error to Sentry?
+            print('Invalid response: %s' % body)
+            return helper.error(ERR_INVALID_RESPONSE)
 
-        # a domain may not yet be configured as this could be the setup flow
-        if self.domain and self.domain != data['domain']:
-            return helper.error(ERR_INVALID_DOMAIN)
+        if not data.get('data').get('email'):
+            #TODO(cnuss): Propagate this error to Sentry?
+            print('Invalid response: %s' % body)
+            return helper.error(ERR_INVALID_RESPONSE)
 
-        helper.bind_state('user', data)
+        domain = extractDomain(data.get('data').get('email'))
 
-        return helper.next_step()
+        if self.domain and self.domain != domain:
+            return helper.error(ERR_INVALID_DOMAIN % (domain, self.domain))
+
+        helper.bind_state('user', data.get('data'))
```

Comment:
So data['domain'] used to exist here, but because of this it no longer does. I guess you could change the build_config to extract domain, but the alternatively is to also grab it here and bind it in the state